files/image_config/reboot-cause/determine-reboot-cause

```diff
@@ -121,6 +126,66 @@ def find_hardware_reboot_cause():
     return hardware_reboot_cause
 
 
+def find_platform_reboot_info():
+    additional_reboot_info = None
+    # Until all platform vendors have provided sonic_platform packages,
+    # if there is no sonic_platform package installed, we don't provide
+    # additional reboot info.
+    try:
+        import sonic_platform
+
+        platform = sonic_platform.platform.Platform()
+
+        chassis = platform.get_chassis()
+
+        additional_reboot_info = chassis.get_reboot_info()
+
+    except ImportError as err:
+        log_warning("sonic_platform package not installed. Unable to retrieve platform reboot info.")
+
+    if additional_reboot_info:
+        log_info("Platform api indicates additional reboot information {}".format(additional_reboot_info))
+    else:
+        log_info("No additional reboot info from platform api")
+
+    return additional_reboot_info
+
+
+def create_prev_cause_history_file_and_symlink(previous_reboot_cause, comment):
+    # Create reboot-cause-#time#.json under history directory
+    FILE_GEN_TIME = str(datetime.datetime.now().strftime('%Y_%m_%d_%H_%M_%S'))
+    REBOOT_CAUSE_HISTORY_FILE = os.path.join(REBOOT_CAUSE_HISTORY_DIR, "reboot-cause-{}.json".format(FILE_GEN_TIME))
+
+    # resultant dictionary 
+    reboot_cause_dict = {} 
+    reboot_cause_dict['gen_time'] = FILE_GEN_TIME
+    if re.search(r'User issued', previous_reboot_cause):
+        # Match with "User issued 'reboot' command [User: arlakshm, Time: Thu Oct  1 20:11:50 UTC 2020]"
+        pattern = 'User issued \'(.*)\' command \[User: (.*), Time: (.*)\]'
+        match = re.search(pattern, previous_reboot_cause)
```

Comment:
It seems like this regex is unnecessary. `find_software_reboot_cause()` currently produces a nicely formatted string for display. But since you're changing things to store a JSON file instead, maybe you should modify `find_software_reboot_cause()` to return a dictionary, then you won't need to use regex here. There's no point in creating a user-readable string if it's not going to be displayed anywhere.

Reply:
The Software reboot-cause is from "reboot-cause.txt" which was written by reboot script. `find_software_reboot_cause()` doesn't do any processing to create a user-readable string.

Reply:
True, `find_software_reboot_cause()` just appends "First boot of SONiC ..." to the string if applicable. I guess we can keep this for now, but we might want to change the reboot script to write a "reboot-cause.json" file instead, then we can remove all this regex.

Reply:
Yes we can work on reboot script part later since we need to work on the reboot script to support some other features but not in this PR.